Add new navigation pattern for Value Tracking

ryzngard opened the pull request:
Value tracking originally used the standard WPF navigation pattern. This would also navigate to the source line based on selection.

This improves the design by manually handling keyboard navigation for most cases. By default, if an item is selected it no longer navigates to source. There are conditions where this is different:

| Shortcut | Expand Behavior | Navigates To Source | Previous or Next |
| --------- | ---------------- | ---------------------- | --------------------- |
| Arrow Down | False | True| Next |
| Arrow Up | False | True| Next |
| Arrow Left | Collapse Only | False | Neither |
| Arrow Right | Expand Only | False | Neither |
| Space | Toggle | False | Neither |
| Enter | False | True | Neither |
| F8 | Expand Only | True | Next |
| Shift + F8 | False | True | Previous |
| Click | False | True| Neither |
| Double Click | False | True | Neither |

Resolves #53821

davidwengier commented on src/VisualStudio/Core/Def/ValueTracking/TreeViewItemBase.cs:
```diff
@@ -33,6 +40,97 @@ public bool IsLoading
+                        item.Parent = parent;
+                    }
+                }
+            };
```
Call me old, but a 20 line lambda with a local function should probably just be a method 😛

ryzngard replied:
maybe... you don't always know when you start writing it :P

davidwengier commented on src/VisualStudio/Core/Def/ValueTracking/ValueTrackingTree.xaml.cs:
```diff
@@ -32,9 +32,97 @@ public ValueTrackingTree(ValueTrackingTreeViewModel viewModel)
+        {
+            if (sender is TreeViewItemBase viewModel)
+            {
+                SelectItem(viewModel, true);
```
Shouldn't this also expand, based on your table?

ryzngard replied:
Actually, I think the table is wrong here. Imo clicking on an item shouldn't expand at all. Open to feedback on that behavior though

ryzngard replied:
Updated table to reflect this behavior as written

davidwengier replied:
As the person who made double click on a project file open the editor, and not expand/collapse, I get the logic of it and support you, but you should probably expect feedback too :)

Cosifne replied:
I believe double-clicking in Test Explorer would do an expansion. And I just know solution explorer does a different thing😮

ryzngard commented on src/VisualStudio/Core/Def/ValueTracking/TreeItemViewModel.cs:
```diff
@@ -70,6 +70,7 @@ internal class TreeItemViewModel : TreeViewItemBase
             Workspace = workspace;
             DocumentId = documentId;
 
+
```
every... single... time...

mikadumont commented:
I do think arrow up, arrow down, and single click should navigate to source. We should keep behavior consistent with other results window like Find All References and shouldn't add keyboard friction causing the experience to be less seamless for users.

ryzngard commented:
I've changed the table to match Mika's request and updated the code. If we get more user feedback on wanting this to be different we will reevaluate.
